src/renderer/components/Runtime.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect } from 'react'
+
+import useMessage from '../hooks/useMessage'
+import useProcessQr from '../hooks/useProcessQr'
+import { clearNotificationsForChat } from '../system-integration/notifications'
+import { runtime } from '../runtime'
+import { selectedAccountId } from '../ScreenController'
+
+import type { PropsWithChildren } from 'react'
+
+/**
+ * Helper component to hook React methods into the external "runtime". This
+ * allows us to interact with the underlying Electron runtime and
+ * operating system.
+ */
+export default function Runtime({ children }: PropsWithChildren<{}>) {
```

Comment:
we should maybe rename this to avoid confusion with the actual runtime interface, make "RuntimeConnection" or similar, as this only sets some specific callbacks for the runtime

Reply:
or RuntimeConnection or RuntimeConnectionController or whatever that's different enough from the interface name

Reply:
I've changed it to `RuntimeAdapter` now, it's an "adapter" for the DeltaChat Desktop React implementation to the runtime interface